migrations/V109__Alter_Team_Role_To_Array.sql

```diff
@@ -0,0 +1,40 @@
+DROP TRIGGER collaborator_lead_req_update ON plan_collaborator;
+DROP TRIGGER collaborator_lead_req_delete ON plan_collaborator;
+DROP FUNCTION collaborator_role_check_trigger();
+
+ALTER TABLE plan_collaborator RENAME COLUMN team_role TO team_roles;
+
+-- Alter the team_roles column to be an array of team_role enum and convert existing values to array format
+ALTER TABLE plan_collaborator
+  ALTER COLUMN team_roles TYPE team_role[] USING ARRAY[team_roles];
+
+CREATE FUNCTION collaborator_role_check_trigger() RETURNS TRIGGER AS $role_check$
+BEGIN
+  -- Check if 'MODEL_LEAD' exists in the team_roles array for the given model_plan_id
+  IF (
+       SELECT count(*)
+       FROM plan_collaborator
+       WHERE 'MODEL_LEAD' = ANY(team_roles) AND model_plan_id = OLD.model_plan_id
+     ) < 2 THEN
+    RAISE EXCEPTION 'There must be at least one MODEL_LEAD assigned to each model plan';
+  END IF;
+
+  IF (TG_OP = 'DELETE') THEN
+    RETURN OLD;
+  ELSE
+    RETURN NEW;
+  END IF;
+END
+$role_check$ LANGUAGE plpgsql;

```

Comment:
![image](https://github.com/CMSgov/mint-app/assets/15203744/6862f86c-538c-4eba-b5b8-a48555288c85)

Something isn't working correctly with this function locally. When I attempt to modify a collaborator (only collaborator on the plan, currently only has `["MODEL_LEAD"]` as their roles, I get this error even though the target roles also contain `["MODEL_LEAD"]`

However, I _don't_ get that error when the _only_ role I target is `"MODEL_LEAD"` - seems to be something with the `newRoles` array containing things that aren't `"MODEL_LEAD"` (seen below)

![image](https://github.com/CMSgov/mint-app/assets/15203744/aa151f21-f84c-4058-83bf-6da5e005adb6)